wallet/udb/testdata/v7.go

```diff
@@ -0,0 +1,214 @@
+	if err != nil {
+		return err
+	}
+
+	return walletdb.Update(db, func(dbtx walletdb.ReadWriteTx) error {
+		amgrns := dbtx.ReadWriteBucket([]byte("waddrmgr"))
+		txmgrns := dbtx.ReadWriteBucket([]byte("wtxmgr"))
+		err := amgr.Unlock(amgrns, privPass)
+		if err != nil {
+			return err
+		}
+
+		privKey, _ := secp256k1.PrivKeyFromBytes(privKey)
+		wif, err := dcrutil.NewWIF(privKey, chainParams, chainec.ECTypeSecp256k1)
+		if err != nil {
+			return err
+		}
+		maddr, err := amgr.ImportPrivateKey(amgrns, wif)
+		if err != nil {
+			return err
+		}
+		addr := maddr.Address()
+
+		// Add a block
+		prevBlock := chainParams.GenesisHash
+		buf := bytes.Buffer{}
+		err = (&wire.BlockHeader{
+			Version:      1,
+			PrevBlock:    *prevBlock,
+			StakeVersion: 1,
+			VoteBits:     1,
+			Height:       uint32(1),
+		}).Serialize(&buf)
+		if err != nil {
+			return err
+		}
+
+		headerData := udb.BlockHeaderData{
+			BlockHash: chainhash.Hash{31: byte(1)},
+		}
+		copy(headerData.SerializedHeader[:], buf.Bytes())
+		err = txmgr.ExtendMainChain(txmgrns, &headerData)
+		if err != nil {
+			return err
+		}
+
+		blockMeta := &udb.BlockMeta{
+			Block: udb.Block{
+				Hash:   headerData.BlockHash,
+				Height: int32(1),
+			},
+			Time: epoch,
+		}
+
+		// Add 6 sstxchange outputs (3 mined, 3 unmined)
+		for count := 1; count < 7; count++ {
+			msgTx := wire.NewMsgTx()
+			pkScript, err := txscript.PayToSStxChange(addr)
+			if err != nil {
+				return fmt.Errorf("cannot create txout script: %s", err)
+			}
+
+			msgTx.AddTxOut(wire.NewTxOut(int64(dcrutil.Amount(1*count)), pkScript))
+			// set tx expiry
+			msgTx.Expiry = 500
+
+			rec, err := udb.NewTxRecordFromMsgTx(msgTx, epoch)
+			if err != nil {
+				return err
+			}
+
+			if count%2 == 0 {
+				err = txmgr.InsertMinedTx(txmgrns, amgrns, rec, &headerData.BlockHash)
+				if err != nil {
+					return err
+				}
+				err = txmgr.AddCredit(txmgrns, rec, blockMeta, 0, true, 0)
+				if err != nil {
+					return err
+				}
+			} else {
+				err = txmgr.InsertMemPoolTx(txmgrns, rec)
+				if err != nil {
+					return err
+				}
+
+				err = txmgr.AddCredit(txmgrns, rec, nil, 0, true, 0)
+				if err != nil {
+					return err
+				}
+			}
+		}
+
+		// add a regular tx
+		msgTx := wire.NewMsgTx()
+		p2pkh, err := dcrutil.DecodeAddress("TsmbdTvkBrXqAhy6W4nror8XERdgLdDW8DM")
+		if err != nil {
+			return fmt.Errorf("failed to decode address: %s", err)
+		}
+		pkScript, err := txscript.PayToAddrScript(p2pkh)
+		if err != nil {
+			return fmt.Errorf("cannot create txout script: %s", err)
+		}
+
+		// using second mined tx (value of 2 DCR) as input
+		msgTx.AddTxIn(wire.NewTxIn(&wire.OutPoint{Index: 1}, nil))
+		msgTx.AddTxOut(wire.NewTxOut(int64(dcrutil.Amount(1)), pkScript))
+		// set tx expiry
+		msgTx.Expiry = 500
+		rec, err := udb.NewTxRecordFromMsgTx(msgTx, epoch)
+		if err != nil {
+			return err
+		}
+
+		err = txmgr.InsertMemPoolTx(txmgrns, rec)
+		if err != nil {
+			return err
+		}
+
+		return nil
```

Comment:
One thing that is missing from this db artifact is that all of there transactions added have expiry set, there's no tx that doesn't.  We should add one more tx that does not set expiry and make sure that unlike the rest, it is not marked as having a set expiry in the upgrade test.